Case Study on Measures of Central Tendency and Dispersion

An Institution wishes to find out their student’s ability in maths, reading and writing skills. The Institution wants to do an exploratory study to check the following information.

1. Find out how many males and females participated in the test. 
2. What do you think about the students' parental level of education? 
3. Who scores the most on average for math, reading and writing based on ● Gender ● Test preparation course
4. What do you think about the scoring variation for math, reading and writing based on ● Gender ● Test preparation course 
5. The management needs your help to give bonus points to the top 25% of students based on their maths score, so how will you help the management to achieve this.

In [1]:
import numpy as np
import pandas as pd

# Reading the Data

In [7]:
dfScores = pd.read_csv('studentsperformance.csv')
dfScores.head()

gender race/ethnicity parental level of education         lunch  \
0  female        group C            some high school  free/reduced   
1  female        group B                 high school  free/reduced   
2  female        group B            some high school  free/reduced   
3  female        group B                some college      standard   
4  female        group C                some college  free/reduced   

  test preparation course  math score  reading score  writing score  
0                    none           0             17             10  
1                    none           8             24             23  
2                    none          18             32             28  
3                    none          11             38             32  
4                    none          22             39             33

# 1. Find out how many males and females participated in the test. 

In [3]:
dfScores.groupby("gender").size()

gender
female    518
male      482
dtype: int64

In [5]:
#518 Females and 482 Males participated in the test

# 2. What do you think about the students' parental level of education? 

In [8]:
dfScores.groupby("parental level of education").size()

parental level of education
associate's degree    222
bachelor's degree     118
high school           196
master's degree        59
some college          226
some high school      179
dtype: int64

# 59 parents are with master's degree, 118 with bachelor's degree,  222 parents are with associate's degree , 226 - some college , 196 high school and 179 some high school

# 3. Who scores the most on average for math, reading and writing based on ● Gender ● Test preparation course

In [57]:
dfAgg = dfScores.groupby(["gender","test preparation course"]).agg({'math score':['mean']}).round(2)
dfAgg

math score
                                     mean
gender test preparation course           
female completed                    67.11
       none                         61.32
male   completed                    72.99
       none                         66.47

# Female students who had completed the test preparation course has more math score on average than test preparation course none
# Among males also the same
# Male students with test preparation course completed scores more on maths than females

In [58]:
dfAgg = dfScores.groupby(["gender","test preparation course"]).agg({'reading score':['mean']}).round(2)
dfAgg

reading score
                                        mean
gender test preparation course              
female completed                       77.38
       none                            69.96
male   completed                       70.79
       none                            62.58

# Female students who had completed the test preparation course has more reading score on average than test preparation course none
# Among males also the same
# Female students with test preparation course completed scores more on reading than male students

In [59]:
dfAgg = dfScores.groupby(["gender","test preparation course"]).agg({'writing score':['mean']}).round(2)
dfAgg

writing score
                                        mean
gender test preparation course              
female completed                       78.79
       none                            68.98
male   completed                       70.34
       none                            59.55

# Female students who had completed the test preparation course has more writing  score on average than test preparation course none
# Among males also the same
# Female students with test preparation course completed scores more on writing  than male students

# 4. What do you think about the scoring variation for math, reading and writing based on ● Gender ● Test preparation course

In [ ]:
Students who completed the test preparation course scores more on math,reading and writing in both the genders
Female students got more on reading and writing where as male students got more on math.
The test preparation course has a positive effect on both Male and female students


In [60]:
 dfScores.groupby(["gender","test preparation course"]).var()

math score  reading score  writing score
gender test preparation course                                          
female completed                208.173913     161.175546     149.836897
       none                     272.602767     214.241220     225.495172
male   completed                197.098133     185.297987     166.098233
       none                     205.513600     184.843553     181.394687

# 5. The management needs your help to give bonus points to the top 25% of students based on their maths score, so how will you help the management to achieve this.

In [142]:
dfScores[dfScores['math score'] >= dfScores.quantile(.75)['math score']]

gender race/ethnicity parental level of education         lunch  \
414  female        group A                 high school  free/reduced   
415  female        group B             master's degree  free/reduced   
416  female        group B           bachelor's degree  free/reduced   
417  female        group B             master's degree      standard   
418  female        group B                 high school      standard   
..      ...            ...                         ...           ...   
995    male        group E                some college      standard   
996    male        group A                some college      standard   
997    male        group D                some college      standard   
998    male        group E          associate's degree  free/reduced   
999    male        group E           bachelor's degree      standard   

    test preparation course  math score  reading score  writing score  
414               completed          77             88             85  
415               completed          77             97             94  
416                    none          77             85             87  
417                    none          77             90             84  
418               completed          77             82             89  
..                      ...         ...            ...            ...  
995               completed          99             87             81  
996               completed         100             96             86  
997               completed         100             97             99  
998               completed         100            100             93  
999               completed         100            100            100  

[255 rows x 8 columns]

In [ ]:
in the above result top 25% is taken from the last quartile.
students with math score greater than or equal to 77 are taken out.
so it gives 255 students from the 1000 as top 25%